0010-R-components-healthcheck.md

```diff
@@ -0,0 +1,477 @@
+# Components Healthcheck In Dapr
+
+* Author(s): Deepanshu Agarwal (@DeepanshuA)
+* State: Draft
+* Updated: 07/20/2023
+
+## Overview
+As a user, I should be able to enquire the health of dapr system, which is possible currently - but this health check should also include health of connected dapr components as well.
+
+## Tenets
+1. It should be extensible i.e. tomorrow if any feature to check health of Actors etc. is required, it should not require a new endpoint.
+2. It should not give any false positives or false negatives.
+
+## Current Scenario
+There are many components in Dapr which don't yet implement Ping. 
```

Comment:
Comment from Alessandro (@ItalyPaleAle ): That said, we should review components that don't implement Ping, and see if adding it would be useful.

Ref: https://hackmd.io/MaNpUYRyQqe-0eqCsSUiIg

Reply:
Agreed